src/validation.cairo

```diff
@@ -120,11 +125,25 @@ pub impl TransactionValidatorImpl of TransactionValidator {
     }
 }
 
+
 fn block_hash(self: @ChainState, block: @Block, merkle_root: Hash) -> Result<Hash, ByteArray> {
-    // TODO: implement
-    Result::Ok(Default::default())
+    let header = block.header;
+    let mut header_data = ArrayTrait::<u32>::new();
+    header_data.append(*header.version);
+    header_data.append(*header.time);
+    header_data.append(*header.bits);
+    header_data.append(*header.nonce);
+    header_data.append_span(merkle_root.value.span());
```

Comment:
Fields are serialized in a wrong order, see block header structure https://learnmeabitcoin.com/technical/block/

Reply:
Got it. Interesting the fact that the order matters.